app/views/govuk_publishing_components/components/docs/attachment.yml

```diff
@@ -0,0 +1,49 @@
+name: Attachment (experimental)
+description: Displays a link to download an attachment and metadata about the file
+body: |
+  This component is marked as experimental as it is part of a drive to provide
+  a consistent place for attachment presentation logic (currently this logic is
+  within [Govspeak][] and [Whitehall][]). The API for this
+  component may change during this process.
+
+  It is used to show data about an attachment document to GOV.UK content.
+
+  [Govspeak]: https://github.com/alphagov/govspeak/blob/c3a0742c87537a371108d097cea23688efa90a14/lib/govspeak/presenters/attachment_presenter.rb
+  [Whitehall]: https://github.com/alphagov/whitehall/blob/5c760eea912320c5a80ef59791df47161d889f2f/app/helpers/document_helper.rb
+shared_accessibility_criteria:
+- link
+examples:
+  default:
+    data:
+      attachment:
+        title: "Department for Transport information asset register"
+        url: https://assets.publishing.service.gov.uk/government/uploads/system/uploads/attachment_data/file/747661/department-for-transport-information-asset-register.csv
+        filename: department-for-transport-information-asset-register.csv
```

Comment:
What's `filename` for? It doesn't seem to be used anywhere?

Reply:
We can use the filename to determine the content type of the attachment: https://github.com/alphagov/govuk_publishing_components/blob/dc8595d2a7d6a2404f85666a387b8f466ae960c3/lib/govuk_publishing_components/presenters/attachment.rb#L27

Reply:
Ah, fair enough, thanks.